transports/quic/src/transport.rs

```diff
@@ -202,6 +212,8 @@ impl<P: Provider> Transport for GenTransport<P> {
                 errored.push(*key);
             }
         }
+        self.waker = Some(cx.waker().clone());
```

Comment:
Can we pair this with the `Poll::Pending` below?

Reply:
I am in favor of the above suggestion. Something along the lines of:


``` diff
-       self.waker = Some(cx.waker().clone());


        for key in errored {
            // Endpoint driver of dialer crashed.
            // Drop dialer and all pending dials so that the connection receiver is notified.
            self.dialer.remove(&key);
        }
-       match self.listeners.poll_next_unpin(cx) {
-           Poll::Ready(Some(ev)) => Poll::Ready(ev),
-            _ => Poll::Pending,
-       }
+        if let Poll::Ready(Some(ev)) = self.listeners.poll_next_unpin(cx) {
+           return Poll::Ready(ev);
+       }

+       self.waker = Some(cx.waker().clone());
+       Poll::Pending
```

Reply:
Done in 7cdaee7.